contracts/contracts/multisig/Multisig.cairo

```diff
@@ -474,24 +449,104 @@ func execute_transaction{
     return (response_len=response.retdata_size, response=response.retdata)
 end
 
+# Sets number of required confirmations. The only way this can be invoked
+# is via a recursive call from execute_transaction -> set_confirmations_required.
+@external
+func set_confirmations_required{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    confirmations_required : felt
+):
+    require_multisig()
+
+    const lower_bound = 1
+
+    let (owners_len) = _owners_len.read()
+    with_attr error_message("invalid number of required confirmations"):
+        assert_in_range(confirmations_required, lower_bound, owners_len + 1)
+    end
+
+    _confirmations_required.write(confirmations_required)
+    ConfirmationsSet.emit(confirmations_required)
+    return ()
+end
+
+# Sets the owners field on the multisig. The only way this can be invoked
+# is via a recursive call from execute_transaction -> set_owners.
+# Number of required confirmations is decreased in case its larger than
+# number of new owners.
+@external
+func set_owners{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+    owners_len : felt, owners : felt*
+):
+    alloc_locals
+
+    require_multisig()
+
+    with_attr error_message("invalid num of owners"):
+        assert_not_zero(owners_len)
+    end
+
+    with_attr error_message("owners has to be unique"):
```

Comment:
The other time this is checked on L334, the error message will be "owners not unique".

Let's try to extract common logic and reuse it so we avoid inconsistencies like this.